pkg/controller/invocation/controller.go

```diff
@@ -115,28 +121,7 @@ func (cr *Controller) Init(sctx context.Context) error {
 	}
 
 	// process evaluation queue
-	pool := gopool.New(maxParallelExecutions)
-	go func(ctx context.Context) {
-		queue := cr.evalQueue.Chan()
-		for {
-			select {
-			case eval := <-queue:
-				func() {
-					ee := eval
-					err := pool.Submit(ctx, func() {
-						controller.EvalQueueSize.WithLabelValues("invocation").Dec()
-						cr.Evaluate(ee.ID())
-					})
-					if err != nil {
-						log.Errorf("failed to submit invocation %v for execution", eval.ID())
-					}
-				}()
-			case <-ctx.Done():
-				log.Info("Evaluation queue listener stopped.")
-				return
-			}
-		}
-	}(ctx)
+	go wait.Until(cr.runWorker, time.Second, sctx.Done())
```

Comment:
Actually `runWorker` is `for{}-select` function. It will wait for queue key endless. So simple calling will lead deadloop.

`wait.Until()` will not loop the `runWorker`, unless `runWorker` be completed. Actually `runWorker` is endless loop. So we can say `wait.Until()` perform that if `runWorker` complete, it means the program fall in error, and then `wait.Until()` restart it, to ensure program going on.

We can see that K8S example or other similar projects implement it as same.

Reply:
Okay, thanks for explaining 👍